google-cloud-firestore/lib/google/cloud/firestore/convert.rb

```diff
@@ -277,26 +269,21 @@ def writes_for_set_merge doc_path, data, field_paths, allow_empty
               end
             end
 
-            if data.any? || field_paths.any? || (allow_empty && field_paths_and_values.empty?)
-              writes << Google::Cloud::Firestore::V1::Write.new(
-                update: Google::Cloud::Firestore::V1::Document.new(
-                  name: doc_path,
-                  fields: hash_to_fields(data)),
-                update_mask: Google::Cloud::Firestore::V1::DocumentMask.new(
-                  field_paths: field_paths.map(&:formatted_string).sort)
-              )
-            end
-
-            if field_paths_and_values.any?
-              writes << transform_write(doc_path, field_paths_and_values)
-            end
-
-            writes
+            doc = Google::Cloud::Firestore::V1::Document.new(
+              name:   doc_path,
+              fields: hash_to_fields(data)
+            )
+            doc_mask = Google::Cloud::Firestore::V1::DocumentMask.new(
+              field_paths: field_paths.map(&:formatted_string).sort
+            )
+            Google::Cloud::Firestore::V1::Write.new(
+              update:            doc,
+              update_mask:       doc_mask,
+              update_transforms: field_transforms(field_paths_and_values)
+            )
```

Comment:
@schmidt-sebastian Please review lines 272-283 to confirm that the `Write` looks correct for a set merge operation.

Reply:
LGTM